### PR TITLE
Implemented Code Review comments, Copy code api changes, stackblitz intergration for preview

### PR DESCRIPTION
**Pull request by vinaydhakretutanota**

### CODE REVIEW

File: projects/external-components/src/lib/primeng-selectors/components/generate-with-ai/generate-with-ai.component.html [hunk 1]
```diff
@@ -5,12 +5,21 @@
...
           <div class="button-code">
-            <div class="language-code">typescript</div>
-            <div class="code-import-button" (click)="copycode( part.content)">
-              <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="none" viewBox="0 0 24 24" class="icon-sm">
-                <path fill="currentColor" fill-rule="evenodd" d="M7 5a3 3 0 0 1 3-3h9a3 3 0 0 1 3 3v9a3 3 0 0 1-3 3h-2v2a3 3 0 0 1-3 3H5a3 3 0 0 1-3-3v-9a3 3 0 0 1 3-3h2zm2 2h5a3 3 0 0 1 3 3v5h2a1 1 0 0 0 1-1V5a1 1 0 0 0-1-1h-9a1 1 0 0 0-1 1zM5 9a1 1 0 0 0-1 1v9a1 1 0 0 0 1 1h9a1 1 0 0 0 1-1v-9a1 1 0 0 0-1-1z" clip-rule="evenodd"></path>
-              </svg>
-               Copy Code</div>
+            <div class="language-code">Typescript</div>
+            <div class="button-actions">
+              <div class="code-import-button" (click)="copycode(part.content)" pTooltip="Copy Code to Clipboard" tooltipPosition="top">
```
Comment (ui-daddy) [inline projects/external-components/src/lib/primeng-selectors/components/generate-with-ai/generate-with-ai.component.html:10]: Add a functionality to change label of button from copy code to copied for few seconds.

File: projects/external-components/src/lib/primeng-selectors/components/generate-with-ai/generate-with-ai.component.ts [hunk 2]
```diff
@@ -108,24 +128,26 @@ export class GenerateWithAiComponent extends CommonExternalComponent implements
...
-        if (action.actionType === "SET_SHARED_DATA" && action.sharedData && action.sharedData.length) {
-          action.sharedData.forEach((shareDataObj: any) => {
-            if (shareDataObj.staticData === "$copyCodeDatas$") {
-              shareDataObj.staticData = this.messageData;
-            }
-          });
-        }
+    const copyCodeEvent = this.fieldObj.events?.find((obj: { event: string; }) => obj.event === "copycode");
+    if (copyCodeEvent) {
+      copyCodeEvent.actions.forEach((action: any) => {
```
Comment (ui-daddy) [inline projects/external-components/src/lib/primeng-selectors/components/generate-with-ai/generate-with-ai.component.ts:137]: Make a new createComponent event. We can keep copyCode event as it is.

File: projects/external-components/src/lib/primeng-selectors/constant/stackblitz-constant.ts [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    "@angular/core": "14.1.3",
+    "@angular/forms": "14.1.3",
+    "@angular/platform-browser": "14.1.3",
+    "@angular/platform-browser-dynamic": "14.1.3",
+    "@angular/router": "14.1.3",
+    "@angular/cdk": "^14.1.3",
+    "rxjs": "~6.6.0",
+    "tslib": "^2.0.0",
+    "zone.js": "~0.11.4",
+    "chart.js": "^4.1.1",
```
Comment (ui-daddy) [inline projects/external-components/src/lib/primeng-selectors/constant/stackblitz-constant.ts:202]: remove chart.js, xlsx.js, file-saver, jspdf, jspdf-autotable

**Review by ui-daddy — APPROVED** [review]
Preview functionality added for AI component

File: projects/external-components/src/lib/primeng-selectors/constant/stackblitz-constant.ts (hunk 3 above)
Comment (vinaydhakretutanota) [reply projects/external-components/src/lib/primeng-selectors/constant/stackblitz-constant.ts:202]: fixed

File: projects/external-components/src/lib/primeng-selectors/components/generate-with-ai/generate-with-ai.component.html (hunk 1 above)
Comment (vinaydhakretutanota) [reply projects/external-components/src/lib/primeng-selectors/components/generate-with-ai/generate-with-ai.component.html:10]: done

File: projects/external-components/src/lib/primeng-selectors/components/generate-with-ai/generate-with-ai.component.ts (hunk 2 above)
Comment (vinaydhakretutanota) [reply projects/external-components/src/lib/primeng-selectors/components/generate-with-ai/generate-with-ai.component.ts:137]: Fixed

**Pull request merged**